geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/SqlHandler.java

```diff
@@ -54,6 +60,13 @@ public SqlHandler(InternalCache cache, String regionName,
     this.tableMetaDataManager = tableMetaDataManager;
     this.regionMapping = getMappingForRegion(configService, regionName);
     this.dataSource = getDataSource(dataSourceFactory, this.regionMapping.getDataSourceName());
+    try {
```

Comment:
move this try/catch to compareTableMetadataWithMapping. I'm not even sure you need this extra exception in that method. Seems like when that method throws it can include all the info in the message.

Reply:
We also need to handle an SQL exception which could be thrown by the MetaDataManager when retrieving the view. We could look at ways to make this more specific.